plugins/kubernetes-backend/src/types/types.ts

```diff
@@ -14,145 +14,22 @@
  * limitations under the License.
  */
 
-import {
-  ExtensionsV1beta1Ingress,
-  V1ConfigMap,
-  V1Deployment,
-  V1HorizontalPodAutoscaler,
-  V1Pod,
-  V1ReplicaSet,
-  V1Service,
-} from '@kubernetes/client-node';
-import { Entity } from '@backstage/catalog-model';
+import type {
+  FetchResponse as FetchResponseCommon,
+  KubernetesFetchError as KubernetesFetchErrorCommon,
+} from '@backstage/plugin-kubernetes-common';
 
-export interface ClusterDetails {
-  name: string;
-  url: string;
-  authProvider: string;
-  serviceAccountToken?: string | undefined;
-}
-
-export interface KubernetesRequestBody {
-  auth?: {
-    google?: string;
-  };
-  entity: Entity;
-}
-
-export interface ClusterObjects {
-  cluster: { name: string };
-  resources: FetchResponse[];
-  errors: KubernetesFetchError[];
-}
-
-export interface ObjectsByEntityResponse {
-  items: ClusterObjects[];
-}
-
-export interface FetchResponseWrapper {
-  errors: KubernetesFetchError[];
-  responses: FetchResponse[];
-}
-
-export type FetchResponse =
-  | PodFetchResponse
-  | ServiceFetchResponse
-  | ConfigMapFetchResponse
-  | DeploymentFetchResponse
-  | ReplicaSetsFetchResponse
-  | HorizontalPodAutoscalersFetchResponse
-  | IngressesFetchResponse
-  | CustomResourceFetchResponse;
-
-// TODO fairly sure there's a easier way to do this
-
-export type KubernetesObjectTypes =
-  | 'pods'
-  | 'services'
-  | 'configmaps'
-  | 'deployments'
-  | 'replicasets'
-  | 'horizontalpodautoscalers'
-  | 'ingresses'
-  | 'customresources';
-
-export interface PodFetchResponse {
-  type: 'pods';
-  resources: Array<V1Pod>;
-}
-
-export interface ServiceFetchResponse {
-  type: 'services';
-  resources: Array<V1Service>;
-}
-
-export interface ConfigMapFetchResponse {
-  type: 'configmaps';
-  resources: Array<V1ConfigMap>;
-}
-
-export interface DeploymentFetchResponse {
-  type: 'deployments';
-  resources: Array<V1Deployment>;
-}
-
-export interface ReplicaSetsFetchResponse {
-  type: 'replicasets';
-  resources: Array<V1ReplicaSet>;
-}
-
-export interface HorizontalPodAutoscalersFetchResponse {
-  type: 'horizontalpodautoscalers';
-  resources: Array<V1HorizontalPodAutoscaler>;
-}
-
-export interface IngressesFetchResponse {
-  type: 'ingresses';
-  resources: Array<ExtensionsV1beta1Ingress>;
-}
-
-export interface CustomResourceFetchResponse {
-  type: 'customresources';
-  resources: Array<any>;
-}
-
-export interface ObjectFetchParams {
-  serviceId: string;
-  clusterDetails: ClusterDetails;
-  objectTypesToFetch: Set<KubernetesObjectTypes>;
-  labelSelector: string;
-  customResources: CustomResource[];
-}
+export type {
+  KubernetesErrorTypes,
+  KubernetesRequestBody,
+} from '@backstage/plugin-kubernetes-common';
```

Comment:
no reason to re-export, it'll just be confusing - individual files that need these types can import from common directly instead

Reply:
Sure, wanted to minimize the change, bit it's best indeed.